dataflow/src/main/java/org/checkerframework/dataflow/cfg/AbstractCFGVisualizer.java

```diff
@@ -0,0 +1,442 @@
+     * @param workList the queue of blocks to be processed
+     * @param sbGraph the {@link StringBuilder} to store the graph
+     */
+    protected void handleSuccessorsHelper(
+            Block cur, Set<Block> visited, Queue<Block> workList, StringBuilder sbGraph) {
+        ConditionalBlock ccur = ((ConditionalBlock) cur);
+        if (cur.getType() == Block.BlockType.CONDITIONAL_BLOCK) {
+            Block thenSuccessor = ccur.getThenSuccessor();
+            sbGraph.append(
+                    addEdge(
+                            ccur.getId(),
+                            thenSuccessor.getId(),
+                            ccur.getThenFlowRule().toString()));
+            addBlock(thenSuccessor, visited, workList);
+            Block elseSuccessor = ccur.getElseSuccessor();
+            sbGraph.append(
+                    addEdge(
+                            ccur.getId(),
+                            elseSuccessor.getId(),
+                            ccur.getElseFlowRule().toString()));
+            addBlock(elseSuccessor, visited, workList);
+        } else {
+            SingleSuccessorBlock sscur = (SingleSuccessorBlock) cur;
+            Block succ = sscur.getSuccessor();
+            if (succ != null) {
+                sbGraph.append(addEdge(cur.getId(), succ.getId(), sscur.getFlowRule().name()));
+                addBlock(succ, visited, workList);
+            }
+        }
+        if (cur.getType() == Block.BlockType.EXCEPTION_BLOCK) {
+            ExceptionBlock ecur = (ExceptionBlock) cur;
+            for (Map.Entry<TypeMirror, Set<Block>> e : ecur.getExceptionalSuccessors().entrySet()) {
+                TypeMirror cause = e.getKey();
+                String exception = cause.toString();
+                if (exception.startsWith("java.lang.")) {
+                    exception = exception.replace("java.lang.", "");
+                }
+                for (Block b : e.getValue()) {
+                    sbGraph.append(addEdge(cur.getId(), b.getId(), exception));
+                    addBlock(b, visited, workList);
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks whether a block exists in the visited blocks list, and, if not, adds it to the visited
+     * blocks list and the work list.
+     *
+     * @param b the block to check
+     * @param visited the set of blocks that have already been visited
+     * @param workList the queue of blocks to be processed
+     */
+    protected void addBlock(Block b, Set<Block> visited, Queue<Block> workList) {
+        if (!visited.contains(b)) {
+            visited.add(b);
+            workList.add(b);
+        }
+    }
+
+    /**
+     * Helper method to visualize a block.
+     *
+     * @param bb the block
+     * @param analysis the current analysis
+     * @param cbFooter footer for a conditional block
+     * @param osFooter footer for other situations
+     * @param escapeCharacter the escape string to use
```

Comment:
To use for what?  Please document.  I think it might be for left-justification or for centering.
Also, it's weird that this is named escape**Character** when its type is `String`.

Reply:
If I use Char instead of String, then I cannot use it as an argument to StringJoiner. Also I can only use '\n' instead of `System.lineSeparator()`. So I think maybe it is better to select another name.
If you don't mind, could you please give me an advise?

Reply:
You could rename the variable to `escapeString`.  I'm just asking for the name and the type to be consistent.  You can do that by renaming or by changing the type.

Reply:
Thanks for your kind advise! That sounds good, I will do it.